### PR TITLE
Update Regexp prototype @@replace to newest standart

### PR DESCRIPTION
**Pull request by kisbg**
related test-cases has been removed from test262 skip list

Signed-off-by: bence gabor kis <kisbg@inf.u-szeged.hu>

### CODE REVIEW

File: src/runtime/String.cpp [hunk 1]
```diff
@@ -970,7 +970,7 @@ void* UTF16String::operator new(size_t size)
     return GC_MALLOC_EXPLICITLY_TYPED(size, descr);
 }
 
-String* String::getSubstitution(ExecutionState& state, String* matched, String* str, size_t position, StringVector& captures, Value namedCapture, String* replacement)
+String* String::getSubstitution(ExecutionState& state, String* matched, String* str, size_t position, StringVector& captures, Object* namedCapture, String* replacement)
```
Comment (clover2123) [inline src/runtime/String.cpp:973]: Why did you change the type of `namedCapture` parameter?
I think that `Value` type is more suitable because standard also handle it as Value type.

Comment (kisbg) [reply src/runtime/String.cpp]: I  revert my changes on getSubstitution method

**Review by clover2123 — APPROVED** [review]
LGTM

**Comment by kisbg** [comment]
@clover2123  Thank your for the review!

**Review by ksh8281 — APPROVED** [review]
Great!
